hdfs/src/main/scala/akka/stream/alpakka/hdfs/javadsl/HdfsSink.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.hdfs.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.stream.alpakka.hdfs.{HdfsWritingSettings, WriteLog}
+import akka.stream.alpakka.hdfs.scaladsl.{RotationStrategy, SyncStrategy}
+import akka.stream.javadsl
+import akka.util.ByteString
+import akka.{Done, NotUsed}
+import org.apache.hadoop.fs.FileSystem
+import org.apache.hadoop.io.SequenceFile.CompressionType
+import org.apache.hadoop.io.Writable
+import org.apache.hadoop.io.compress.CompressionCodec
+
+object HdfsSink {
+
+  /*
+   * Java API: creates a Sink with [[HdfsFlowStage]] for [[FSDataOutputStream]]
+   *
+   * @param fs HDFS FileSystem
+   * @param syncStrategy Sync Strategy
+   * @param rotationStrategy Rotation Strategy
+   * @param settings Hdfs writing settings
+   */
+  def data(
+      fs: FileSystem,
+      syncStrategy: SyncStrategy,
+      rotationStrategy: RotationStrategy,
+      settings: HdfsWritingSettings
+  ): javadsl.Sink[ByteString, CompletionStage[Done]] =
+    HdfsFlow
+      .data(fs, syncStrategy, rotationStrategy, settings)
+      .toMat(javadsl.Sink.ignore[WriteLog], javadsl.Keep.right[NotUsed, CompletionStage[Done]])
+
+  /*
+   * Java API: creates a Sink with [[HdfsFlowStage]] for [[CompressionOutputStream]]
+   *
+   * @param fs HDFS FileSystem
+   * @param syncStrategy Sync Strategy
+   * @param rotationStrategy Rotation Strategy
+   * @param compressionCodec a class encapsulates a streaming compression/decompression pair.
+   * @param settings Hdfs writing settings
+   */
+  def compressed(
```

Comment:
To reduce the API, you could just have `HdfsSink.data` and let the users connect to `Sink.ignore` for the other cases.

Reply:
Sorry I could not understand. `HdfsSink.data` , `HdfsSink.compressed`, `HdfsSink.sequence` are completely different. What exactly do you mean by letting users connect to `Sink.ignore`

Reply:
I think users needing these cases might as well use the `HdfsFlow` and connect it to `Sink.ignore` themselves.

Reply:
oh, then can we remove Sink implementations?

Reply:
Yes, just keep the most basic ones. Most users discover they'll need to do something after sending/writing to the destination.
Speaking of that, what we most often need is some kind of pass-through, a value untouched by the flow, but available afterwards (eg. Kafka offsets for committing). Have you thought about that? It would require a wrapper for `Writable`, AFAICS.

Reply:
Actually, I was thinking about `pass-through` like Solr connector. I was just not sure to implement. It will be added soon.

Reply:
I would like to consult about it. I was thinking the best strategy for it. Currently, the flow expects `ByteString` as an input. Whenever it rotates the output, it pushes `WriteLog` to the downstream.

If we want to implement `pass-through,` how shall we design it? The dummies way is
```
final case class WriteLog[T](path: String, rotation: Int, passThroughs: Seq[T])
```
However, if we have millions input, keeping this sequence in memory until the flow rotates is super inefficient. 

The second idea is
```
sealed trait OutgoingMessage
final case class RotationMessage(path: String, rotation: Int) extends OutgoingMessage
final case class PassThrough[T](pass: T)  extends OutgoingMessage
```
So flow can push `RotationMessage` when it rotates, and for the rest, it will push `PassThrough`. However, this has a drawback. Let's talk about Kafka example, we write the input to output and send `PassThrough` message with an offset. If something goes wrong in the flow, and if it does not synchronize the output, we will basically fail, but downstream can already commit this offset.

I actually did not like this idea also. Do you have any idea for it?